src/components/Login/Login.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from "react";
+import "./Login.css";
+import { Input } from "@material-ui/core";
+import { Button } from "@material-ui/core";
+import { Link, useNavigate } from "react-router-dom";
+import M from "materialize-css";
+import { useMedia } from "../../MediaContext";
+import { UserAction } from "../../action/userAction";
+import { useDispatch, useSelector } from "react-redux";
+
+function Login() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  let navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const postData = () => {
+    if (
+      !/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(
```

Comment:
Regex Expression can be placed inside in a separate folder and could be named Constants for better code management and readability.